packages/aws-rfdk/lib/lambdas/nodejs/sep-configuration/handler.ts

```diff
@@ -0,0 +1,288 @@
+      const stringConfigs = this.spotFleetRequestToString(resourceProperties.spotFleetRequestConfigurations);
+      const response = await eventPluginRequests.saveServerData(stringConfigs);
+      if (!response) {
+        console.log(`Failed to save spot fleet request with configuration: ${stringConfigs}`);
+      }
+    }
+    if (resourceProperties.spotPluginConfigurations) {
+      const spotEventFleetPluginConfigs = this.spotFleetPluginConfigsToArray(resourceProperties.spotPluginConfigurations);
+      const response = await eventPluginRequests.configureSpotEventPlugin(spotEventFleetPluginConfigs);
+      if (!response) {
+        console.log(`Failed to save Spot Event Plugin Configurations: ${resourceProperties.spotPluginConfigurations}`);
+      }
+    }
+    return undefined;
+  }
+
+  /**
+   * @inheritdoc
+   */
+  /* istanbul ignore next */ // @ts-ignore
+  public async doDelete(physicalId: string, resourceProperties: ISEPConfiguratorResourceProperties): Promise<void> {
+    // Nothing to do -- we don't modify anything.
+    return;
+  }
+
+  private implementsISEPConfiguratorResourceProperties(value: any): boolean {
+    if (!value || typeof(value) !== 'object') { return false; }
+    if (!this.implementsIConnectionOptions(value.connection)) { return false; }
+    if (!this.isValidSFRConfig(value.spotFleetRequestConfigurations)) { return false; }
+    if (!this.isValidSpotPluginConfig(value.spotPluginConfigurations)) { return false; }
+    return true;
+  }
+
+  private implementsIConnectionOptions(value: any): boolean {
+    if (!value || typeof(value) !== 'object') { return false; }
+    if (!value.hostname || typeof(value.hostname) !== 'string') { return false; }
+    if (!value.port || typeof(value.port) !== 'string') { return false; }
+    const portNum = Number.parseInt(value.port, 10);
+    if (Number.isNaN(portNum) || portNum < 1 || portNum > 65535) { return false; }
+    if (!value.protocol || typeof(value.protocol) !== 'string') { return false; }
+    if (value.protocol !== 'HTTP' && value.protocol !== 'HTTPS') { return false; }
+    if (!this.isSecretArnOrUndefined(value.caCertificate)) { return false; }
+    if (!this.isSecretArnOrUndefined(value.passphrase)) { return false; }
+    if (!this.isSecretArnOrUndefined(value.pfxCertificate)) { return false; }
+    return true;
+  }
+
+  private isSecretArnOrUndefined(value: any): boolean {
+    if (value) {
+      if (typeof(value) !== 'string' || !isSecretArn(value)) { return false; }
+    }
+    return true;
+  }
+
+  private isValidSFRConfig(value: any): boolean {
+    if (!value) { return true; }
+    if (typeof(value) !== 'object') { return false; }
+    if (Array.isArray(value)) { return false; }
+    return true;
+  }
+
+  private isValidSpotPluginConfig(value: any): boolean {
+    if (!value) { return true; }
+    if (typeof(value) !== 'object') { return false; }
+    if (Array.isArray(value)) { return false; }
+
+    return true;
+  }
```

Comment:
I added more validations, but they are not called from the `validateInput()`. We will validate the input as the same time when converting some values from strings, to avoid doing validation twice.

Reply:
Looks good